.github/workflows/Tests.yaml

```diff
@@ -433,3 +447,94 @@ jobs:
           # We get occasional random timeouts, repeat tests to see if
           # it is a random timeout or systematic
           ctest -j2 --output-on-failure -L inputfiles --repeat after-timeout:3
+
+  # Release a new version on scheduled or manual events when the tests pass.
+  # Only enable this on the `sxs-collaboration/spectre` repository (not on
+  # forks).
+  release_version:
+    name: Release version
+    if: >
+      github.repository == 'sxs-collaboration/spectre' &&
+        github.ref == 'refs/heads/develop' &&
+        (github.event_name == 'schedule'
+          || (github.event_name == 'workflow_dispatch'
+            && github.event.inputs.release_version != ''))
+    needs:
+      - check_files_and_formatting
+      - doc_check
+      - unit_tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install Python dependencies
+        run: |
+          pip install GitPython~=3.1.11 PyGithub~=1.53 tqdm~=4.51.0
+      # We use the current date as tag name, unless a tag name was specified
+      # as input to the `workflow_dispatch` event
+      - name: Determine release version
+        id: get_version
+        run: |
+          INPUT_RELEASE_VERSION=${{ github.event.inputs.release_version }}
+          RELEASE_VERSION=${INPUT_RELEASE_VERSION:-$(date +'%Y.%m.%d')}
+          echo "Release version is: ${RELEASE_VERSION}"
+          echo "RELEASE_VERSION=$RELEASE_VERSION" >> $GITHUB_ENV
+      - name: Validate release version
+        run: |
+          VERSION_PATTERN="^([0-9]{4})\.([0-9]{2})\.([0-9]{2})(\.[0-9]+)?$"
+          if [[ $RELEASE_VERSION =~ $VERSION_PATTERN ]]; then
+            if [ $(date +'%Y') != ${BASH_REMATCH[1]} ] ||
+            [ $(date +'%m') != ${BASH_REMATCH[2]} ] ||
+            [ $(date +'%d') != ${BASH_REMATCH[3]} ]; then
+              TODAY=$(date +'%Y.%m.%d')
+              echo "'$RELEASE_VERSION' doesn't match current date '$TODAY'"
+            fi
+          else
+            echo "'$RELEASE_VERSION' doesn't match '$VERSION_PATTERN'"
+            exit 1
+          fi
+          if [ $(git tag -l "v$RELEASE_VERSION") ]; then
+            echo "Tag 'v$RELEASE_VERSION' already exists"
+            exit 1
+          fi
+          if [ $(git rev-parse HEAD) == $(git rev-parse origin/release) ]; then
+            echo "Nothing changed since last release $(git describe release)."
```

Comment:
Since we don't plan on doing any extra testing for monthly releases I don't think we should consider any release "better" or "special" compared to any other release.

Reply:
OK, let's go with that and leave the check in.